src/app/qgsoptions.cpp

```diff
@@ -255,14 +255,18 @@ QgsOptions::QgsOptions( QWidget *parent, Qt::WindowFlags fl, const QList<QgsOpti
 
   //local directories to search when loading c++ plugins
   QStringList pathList = mSettings->value( QStringLiteral( "plugins/searchPathsForPlugins" ) ).toStringList();
-  Q_FOREACH ( const QString &path, pathList )
+  if ( !pathList.isEmpty() )
   {
-    QListWidgetItem *newItem = new QListWidgetItem( mListPluginPaths );
-    newItem->setText( path );
-    newItem->setFlags( Qt::ItemIsEditable | Qt::ItemIsEnabled | Qt::ItemIsSelectable );
-    mListPluginPaths->addItem( newItem );
+    Q_FOREACH ( const QString &path, pathList )
```

Comment:
@DelazJ since you are touching this code

```
const QStringList pathList...

for ( const QString &path : pathList )
```

Reply:
@m-kuhn the `if ( !pathList.isEmpty() )` check is then unnecessary? is it implicite?
That said, giving a look to the different list/tree widgets in the file, it's not always used...

Reply:
Yes, if there is nothing in the pathList, the loop will be executed 0 times, implicitly (that's the case with `Q_FOREACH` as well as `for ( : )`

Reply:
pathList is used for different lists in the file, hence setting it as const makes the build fail

Reply:
I see. If you want to make it nice, rename those other variables. If you want this to get merged with as little effort as possible, remove the `const` and use it in each loop as `for ( const QString &path : qgis::as_const( pathList ) )`.